core-web/libs/block-editor/src/lib/extensions/ai-content-prompt/plugins/ai-content-prompt.plugin.ts

```diff
@@ -0,0 +1,186 @@
+    public editor: Editor;
+
+    public element: HTMLElement;
+
+    public view: EditorView;
+
+    public tippy: Instance | undefined;
+
+    public tippyOptions?: Partial<Props>;
+
+    public pluginKey: PluginKey;
+
+    public component?: ComponentRef<AIContentPromptComponent>;
+
+    private $destroy = new Subject<boolean>();
+
+    constructor(props: AIContentPromptViewProps) {
+        const { editor, element, view, tippyOptions = {}, pluginKey, component } = props;
+
+        super(props);
+
+        this.editor = editor;
+        this.element = element;
+        this.view = view;
+
+        this.tippyOptions = tippyOptions;
+
+        this.element.remove();
+        this.element.style.visibility = 'visible';
+        this.pluginKey = pluginKey;
+        this.component = component;
+
+        this.component.instance.buildForm();
+
+        this.component.instance.formValues.pipe(takeUntil(this.$destroy)).subscribe((data) => {
+            this.editor.commands.updateValue(data);
+        });
+
+        this.component.instance.hide.pipe(takeUntil(this.$destroy)).subscribe(() => {
+            this.editor.commands.closeAIPrompt();
+        });
+
+        this.editor.on('focus', this.focusHandler);
+    }
+
+    update(view: EditorView, prevState?: EditorState) {
+        const next = this.pluginKey?.getState(view.state);
+        const prev = prevState ? this.pluginKey?.getState(prevState) : { open: false };
+
+        if (next?.open === prev?.open) {
+            this.tippy?.popperInstance?.forceUpdate();
+
+            return;
+        }
+
+        if (next.open && next.form) {
+            this.component.instance.buildForm();
+        } else {
+            this.component.instance.cleanForm();
+        }
+
+        this.createTooltip();
+
+        next.open ? this.show() : this.hide();
+    }
+
+    createTooltip() {
+        const { element: editorElement } = this.editor.options;
+        const editorIsAttached = !!editorElement.parentElement;
+
+        if (this.tippy || !editorIsAttached) {
+            return;
+        }
+
+        this.tippy = tippy(editorElement.parentElement, {
+            ...this.tippyOptions,
+            ...TIPPY_OPTIONS,
+            content: this.element,
+            onShow: () => {
+                requestAnimationFrame(() => {
+                    this.component.instance.input.nativeElement.focus();
+                });
+            }
+        });
+    }
+
+    focusHandler = () => {
+        this.editor.commands.closeForm();
+        setTimeout(() => this.update(this.editor.view));
+    };
+
+    show() {
+        this.tippy?.show();
+    }
+
+    hide() {
+        this.tippy?.hide();
+        this.editor.view.focus();
+    }
+
+    destroy() {
+        this.tippy?.destroy();
+        this.editor.off('focus', this.focusHandler);
+        this.$destroy.next(true);
+        this.component.destroy();
+        this.component.instance.formValues.unsubscribe();
```

Comment:
the `takeUntil(this.destroy$)` does this for you, but there's something missing. You just need to delete this line and add a `this.destroy$.complete()` that will mark the subject as completed and therefore it will `unsubscribe` and with it all the observables with `takeUntil(this.destroy$)` will unsubscribe as well that way you don't have to call the `unsubscribe` method in all the observables you initialize.

You can check this example. In this blog they use `unsubscribe` instead of `complete`. But is the same thing.
https://dev.to/shane/rxjs-unsubscribe-patterns-with-angular-5goh